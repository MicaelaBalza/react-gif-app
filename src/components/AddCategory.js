import { useState } from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleChange = ({target}) => {
    setInputValue(target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if( inputValue.trim().length > 2) {
      setCategories(cats => [inputValue, ...cats]);
      setInputValue('');
    }
  };

  return (
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          className="search-input"
          type="text"
          value={inputValue}
          placeholder="Buscar Gifs"
          onChange={handleChange}
        />
        <p className="search-icon" onClick={handleSubmit}>{'\u26B2'}</p>
      </form>
  )
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;
