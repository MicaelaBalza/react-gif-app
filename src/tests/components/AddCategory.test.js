import { shallow } from 'enzyme';

import AddCategory from "../../components/AddCategory";

describe('Pruebas en AddCategory', () => {
  let wrapper;
  const setCategories = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<AddCategory setCategories={setCategories}/>)
  });

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    input.simulate('change', { target: { value }});
  });

  test('no debe submittear el formulario si el campo está vacío', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect(setCategories).not.toHaveBeenCalled();
  })
  
  test('debe llamar el setCategories y limpiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = "Digimon";
    input.simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  });
})
