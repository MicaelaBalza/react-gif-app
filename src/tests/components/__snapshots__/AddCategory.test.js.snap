// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory debe mostrarse correctamente 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <input
    className="search-input"
    onChange={[Function]}
    placeholder="Buscar Gifs"
    type="text"
    value=""
  />
  <p
    className="search-icon"
    onClick={[Function]}
  >
    âš²
  </p>
</form>
`;
